### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by GavenElder**
final submission.

### CODE REVIEW

**Review by A-Tanner — CHANGES_REQUESTED** [review]
Missing a comment about garbage collection, otherwise looks ok. 

Some of the methods/functionality in CircularlyLinkedList appears to have been copy-and-pasted from my own codebase and that's really upsetting. 


Create a class library project called "GenericsHomework.". ✔
Create a node class that can contain a value of any type and points to the next node and traversing the next node points back to the first item.
Define the Node class
Include a constuctor that takes a value. (No validation is necessary on the value). ✔
Add a ToString() override that writes out the value's ToString() result. ✔ 
Add a Next property that references the next node or else refers back to itself if there are no other nodes in the list. ✔
The Next property should be non-nullable (careful to follow the non-nullable property guidelines) ✔
The Next property setter should be private. ✔
Add an Append method that takes a value and appends a new Node instance after the current node (by invoking the Next property). ✔
Add a Clear method that effectively removes all items from a list except the current node. Pay attention as to whether you should be concerned with the following:
Whether it is sufficient to only set Next to itself ✔
Whether to set the removed items to circle back on themselves. In other words, whether to close the loop of the removed items. (Provide a test to show why this is required if it is required). ✔
Given there is a circular list of items, provide a comment to indicate whether you need to worry about garbage collection because all the items point to each other and therefore may never be garbage collected. ❌ --> No explanatory comment found
Create an Exists method to test to see if a value exists in the list. ✔
Throw an error on an attempt to Append a duplicate value. (Make sure you test for this case) ✔
You should not rely on any BCL generic classes for your implementation. ✔
Extra Credit
Do one of the following two options (or both if you want extra extra credit) :)

Implement a VennDiagram structure that contains n Circles that only contains homogenous reference types of any type. ❌
Each circle contains n items and each item can belong to one more more Circle instances.
You are not required to use a Node from earlier in the homework for your venn diagram implementation.
You are welcome to use exising BCL generic classes for the extra credit.
Implement Systm.Collections.Generic.ICollection<T> on the Node class ❌
Fundamentals
Ensure you enable:
nullable reference types is enabled ✔
net6 targetted ✔
C# 10.0 ✔
and enabled .NET analyzers for both projects ✔
For this assignment, always use Assert.AreEqual<T>() (the generic version) ✔
All of the above should be unit tested ✔
Choose simplicity over complexity ✔

File: Assignment4/GenericsHomework/CircularlyLinkedList.Tests/CLLTests.cs [hunk 1]
```diff
@@ -0,0 +1,178 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace GenericsHomework.Tests
+{
+    [TestClass]
+    public class CLLTests
+    {
+        int[] IntData = { 1, 2, 3, 4 };
```
Comment (mmwoodfo) [inline Assignment4/GenericsHomework/CircularlyLinkedList.Tests/CLLTests.cs:9]: These variable names don't match the variable names you're using in your tests. Causing your code to not compile.

File: Assignment4/GenericsHomework/GenericsHomework/CircularlyLinkedList.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
+using System.Diagnostics.CodeAnalysis;
+
+namespace GenericsHomework;
+public class CircularlyLinkedList<T>
+{
+    protected Node<T>? _Cursor { get; set; }
```
Comment (mmwoodfo) [inline Assignment4/GenericsHomework/GenericsHomework/CircularlyLinkedList.cs:6]: For the code review, take a look at PR #94 as the code is similar enough that the review is the same...

**Review by mmwoodfo — COMMENTED** [review]
Your code does not compile and therefore cannot be properly graded. Additionally, your code looks similar to another student's codebase.

**Pull request closed without merging**